lib/internal/modules/cjs/loader.js

```diff
@@ -1091,14 +1091,17 @@ Module.prototype._compile = function(content, filename) {
     if (!resolvedArgv) {
       // We enter the repl if we're not given a filename argument.
       if (process.argv[1]) {
-        resolvedArgv = Module._resolveFilename(process.argv[1], null, false);
+        // The resolution may fail if it requires a preload script
```

Comment:
Yes for the first point.

I'm not sure I understand what you mean in the second, can you detail?

Reply:
> I'm not sure I understand what you mean in the second, can you detail?

I meant we should at least `assert(Array.isArray(getOptionValue('--require')))` in the catch block.

Reply:
I'm not convinced - this codepath (where `resolveArgv` is empty and `_resolveFilename` fails but `_compile` is called) is only possible for the preloaded scripts anyway (since otherwise `_compile` won't be called at all if `_resolvedFilename` fails on the real main entry script).

That said, if you still think it's a blocker, I can add it.

Reply:
>  ..is only possible for the preloaded scripts anyway

Yes, exactly, and that's why it should be an assertion - to prevent people from changing this invariant unexpectedly in the future.

Reply:
Fixed 👍

Reply:
I think this comment is redundant now?